packages/demo-cards/claim.gts

```diff
@@ -69,17 +78,36 @@ class Isolated extends Component<typeof Claim> {
     }
   }
 
-  async connectMetamask() {
+  // this was not bound properly here so need to use an arrow function
+  connectMetamask = async () => {
```

Comment:
rule of thumb is that any async that happens in a component should use ember concurrency. It's not a great idea to wire up a button directly to an async function. A better approach is:

```
// I make all my card methods private to not accidentally expose something that downstream authors might construe as API
@action 
private connectMetamask() {  
   this.doConnectMetamask.perform();
}

private doConnectMetamask = restartableTask(async () => {
    ..... the code that you originally had in connectMetamask goes here....
});
```
This also lets you do the fun trick where you can show something in the component while the connection is underway:
```
{{#if this.doConnectMetamask.isRunning}}
   Connecting to Metamask...
{{/if}}
```

Reply:
resolved https://github.com/cardstack/boxel/pull/353/commits/5bfdf2e6c2a73773ac263ed6a1cb39b20785afdd